tcms/testplans/static/testplans/js/get.js

```diff
@@ -63,5 +64,40 @@ function getTestCaseRowContent(rowContent, testCase) {
     );
     automation_indication_element.removeClass(automated_class_to_remove);
 
+    getTestCaseExpandArea(row, testCase);
+
     return row;
 }
+
+function getTestCaseExpandArea(row, testCase) {
+    row.find('.js-test-case-expand-text').html(testCase.text);
+    if (testCase.notes.trim().length > 0) {
+        row.find('.js-test-case-expand-notes').html(testCase.notes);
+    }
+
+    // draw the attachments
+    var uniqueDivCustomId = `js-tc-id-${testCase.id}-attachments`;
+    // set unique identifier so we know where to draw fetched data
+    row.find('.js-test-case-expand-attachments').parent()[0].id = uniqueDivCustomId;
+
+    jsonRPC('TestCase.list_attachments',[testCase.id],function(data) {
+
+        // cannot use instance of row in the callback
+        var ulElement = $(`#${uniqueDivCustomId} .js-test-case-expand-attachments`);
+
+        if (data.length === 0) {
+            ulElement.children().removeClass('hidden');
+            return;
+        }
+
+        for (var i = 0; i < data.length; i++) {
+            //should create new element for every attachment
+            var liElement = document.createElement('li');
+                liElement.className = 'list-group-item';
```

Comment:
IMO add the li element as a template tag (directly under ul for better readability) and just create new children when you have to add attachments.

Reply:
I already have an li element, this which holds "no attachments" message.If I add one more I have to use JS to show/hide this element, i dont think its necessary.

Reply:
With the use of the `<template>` tag you've shown how easy it is to have the entire stricture defined in the templates and manipulate it via JavaScript.

I don't like HTML inside the JavaScript. 

To resolve the problem when you can have multiple elements try using a selector in `.children()` on line 89 above, or just delete all `<li>` elements. I don't care much if we show "No attachments" or not.